05-part_two.Rmd

```diff
@@ -36,37 +36,35 @@ get_max(c("a", "b", "c"))
 
 
 ## Formalised testing
-Beyond in-script tests, an analyst may decide to take a more formal approach to testing if the project warrants it. Software development has dedicated testing professionals (called "testers") and there is plenty of literature about software testing on the internet. Much of this is transferable to code based analysis, but not all of it. There are times when it makes sense to make a distinction between a model (which is generally operated by a technical specialist) and an IT product (e.g. a website, which is used by a broader user base). Shiny apps sit somewhere between a model and an IT product so we'll include some specific comments about these.
+Beyond in-script tests, an analyst may decide to take a more formal approach to testing if the project warrants it. Software development has dedicated testing professionals (called "testers") and there is plenty of literature about software testing on the internet. Much of this is transferable to code based analysis, but not all of it. There are times when it makes sense to make a distinction between a piece of analysis (which is generally operated by a technical specialist) and an IT product (e.g. a website, which is used by a broader user base). Shiny apps sit somewhere between analysis and an IT product so we'll include some specific comments about these.
 
 There are often four types of software testing covered:
 
-* Unit testing (tests the different components or "units" of code using tools such as `testthat` in R).
-* Integration testing (tests the correct flow of data through the components).
-* System testing (tests the overall outputs make sense).
-* Acceptance testing (tests whether or not the work meets its objectives).
+* Unit testing - tests the different components or "units" of code using tools such as `testthat` in R.
+* Integration testing - tests the correct flow of data through the components.
+* System testing - tests the overall outputs make sense.
+* Acceptance testing - tests whether or not the work meets its objectives.
 
-Software testing can make a distinction between functional and non-functional testing too. In the context of an analytical model, a functional test would be one which tests the model produces what its intended to whereas a non-functional test would cover things such as its run-time and interface (i.e. things which affect the experience of using it, but not the numbers that come out).
+Software testing can make a distinction between functional and non-functional testing too. In the context of analysis, a functional test would be one which tests the analysis produces what its intended to whereas a non-functional test would cover things such as its run-time and interface (i.e. things which affect the experience of using it, but not the numbers that come out). For more information, see [here](https://www.guru99.com/levels-of-testing.html) or explore further with help from a search engine.
```

Comment:
Nicky's comment on this section was:  Given that the focus of the remainder of the chapter is on unit testing I think the flow would be better to talk about unit testing and other testing, bundling integration, system and acceptance as they are more about other aspects, as Rachel states in the text.

So, my suggestion is to drop this paragraph but add in the 'For more information.... sentence at the end of the section where you say more about what the other tests relate to but if you think you can address that comment in another way, please go ahead!

Reply:
I have added in another section after the "unit testing" section for some of this content. Having thought a bit about what Nicky was saying, I think this matches up. Also, having made the change I think I prefer it.